File: Unity-THREE-parity-demo-1/OO.Box.js
!function () { "use strict"

const { GameObject, Vector3, PrimitiveType } = window.UnityEngineAnalog

window.OO = window.OO || {} // the Oopish namespace
window.OO.Box = class {
    constructor (absX, absY, absZ) {
        this.el = null;
        this.x = 0;
        this.y = 0;
        this.z = 0;
        this.x = absX;
        this.y = absY;
        this.z = absZ;
        this.el = GameObject.CreatePrimitive(PrimitiveType.Cube);
        this.el.transform.position = new Vector3(this.x, this.y, this.z);
    }
    MoveAbsX (absX) {
        this.x = absX;
        this.el.transform.position = new Vector3(this.x, this.y, this.z);
    }
}

}()

File: Unity-THREE-parity-demo-1/OO.Box.cs
//// Install Mono (mono-project.com), and then compile:
/*
mcs \
-r:/Applications/Unity/Unity.app/Contents/Managed/UnityEngine.dll \
-r:/Applications/Unity/Unity.app/Contents/Managed/UnityEditor.dll \
-sdk:2 `# Unity needs 3.5 or less (mcs only accepts 2 or 4)`\
-target:library \
OO.Box.cs
*/

//// Then make a soft-link to the DLL:
/*
cd /path/to/your/Unity/project/Assets
ln -s /path/to/your/oodotmd/Unity-THREE-parity-demo-1/OO.Box.dll
*/

//// ...now every time the DLL is recompiled, Unity will refresh itself.

using System.Collections;
using UnityEngine;
namespace OO {
    public class Box {
        public GameObject el;
        public float x = 0;
        public float y = 0;
        public float z = 0;
        public Box (float absX, float absY, float absZ) {
            x = absX;
            y = absY;
            z = absZ;
            el = GameObject.CreatePrimitive(PrimitiveType.Cube);
            el.transform.position = new Vector3(x, y, z);
        }
        public void MoveAbsX (float absX) {
            x = absX;
            el.transform.position = new Vector3(x, y, z);
        }
    }
}
